homeassistant/components/homekit/type_remotes.py

```diff
@@ -143,20 +143,23 @@ def __init__(
             serv_input.configure_char(CHAR_CURRENT_VISIBILITY_STATE, value=False)
             _LOGGER.debug("%s: Added source %s", self.entity_id, source)
 
+    def _get_mapped_sources(self, state: State) -> dict[str, str]:
+        """Return a dict of sources mapped to their homekit safe name."""
+        source_list = state.attributes.get(self.source_list_key, [])
+        if self._mapped_sources_list != source_list:
```

Comment:
this check is not ideal, as any key that is cleaned up will now cause this to be re-generated every time.

Reply:
We only hit this logic after the accessory is set up in if the source is missing in `_async_update_input_state`

In that case we are likely in for a full accessory reset and rebuild anyways since the source list has to be exported when the accessory is created.